### ISSUE TITLE
We should capture all file path for in a spec's ID

### ISSUE THREAD

**Issue by shawncao** [varchar-io/nebula#176]
Replace paths_ with a new object, such as `FileSplit` which should capture {file path, offset, length} and these should be part of the spec identifier. @ritvik-statsig

**Comment by shawncao** [comment]
Quote unresolved comment:
```
we may want to include watermark_ in this new object too - think about the case that Time Macro is at the file name, and if you combine multiple files which have different time values, we will need watermark for each file. Unless we guarantee different watermarked files won't be grouped into one spec.
```

**Comment by shawncao** [comment]
`Similarly, we should include all paths in the string result.`

**Comment by shawncao** [comment]
This is resolved in #177
